File: app/src/main/java/net/bonono/rssreader/repository/Repository.java
package net.bonono.rssreader.repository;

import java.util.List;

public interface Repository<T> {
    void transaction(Runnable inTransaction);

    void bindContext(Context context);

    T save(T entity);

    void remove(T entity);
    void remove(Query<T> query);

    int count(Query<T> query);

    List<T> get(Query<T> query);
}

File: app/src/main/java/net/bonono/rssreader/repository/Repository.kt
package net.bonono.rssreader.repository

interface Repository<T> {
    fun transaction(inTransaction: Runnable)

    fun bindContext(context: Context)

    fun save(entity: T): T

    fun remove(entity: T)
    fun remove(query: Query<T>)

    fun count(query: Query<T>): Int

    operator fun get(query: Query<T>): List<T>
}
